generator/mysql/query_set.go

```diff
@@ -28,20 +32,45 @@ ORDER BY table_name;
 		return nil, fmt.Errorf("failed to query %s metadata result: %w", tableType, err)
 	}
 
-	wg := errgroup.Group{}
-	for i := 0; i < len(tables); i++ {
-		i := i
-		wg.Go(func() (err1 error) {
-			tables[i].Columns, err1 = m.GetTableColumnsMetaData(db, schemaName, tables[i].Name)
-			return err1
-		})
+	tableNames := make([]string, len(tables))
+	for i, table := range tables {
+		tableNames[i] = table.Name
+	}
+
+	tableIndexLookup := make(map[string]int, len(tables))
+	for i, table := range tables {
+		tableIndexLookup[table.Name] = i
+	}
+
+	columns, err := m.GetTableColumnsMetaData(db, schemaName, tableNames)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query %s column meta data", err)
+	}
+
+	for _, column := range columns {
+		tableIndex, ok := tableIndexLookup[column.TableName]
+		if !ok {
+			return nil, fmt.Errorf("table %s not found in the metadata for column %v+", column.TableName, column)
+		}
+
+		table := &tables[tableIndex]
+		table.Columns = append(table.Columns, column.Column)
 	}
 
```

Comment:
I think, to solve the N+1 problem, a better approach would be to use jet `qrm` directly. This means merging everything in one query and letting `qrm.Query` group the result. You would also need to add `sql:"primary_key"` tag to `Name` field of `metadata.Table` struct so `qrm` will work.

Reply:
You were right, this is much nicer now.